### PR TITLE
feat(ngbSelect): Select Dropdown for UI Bootstrap

### PR DESCRIPTION
**Pull request by mohammedzamakhan**
Before submitting a pull request, please make sure you have at least performed the following:

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [ ] added/updated any applicable tests.
 - [x] added/updated any applicable API documentation.
 - [x] added/updated any applicable demos.

### CODE REVIEW

**Comment by mohammedzamakhan** [comment]
selectdropdown

**Comment by mohammedzamakhan** [comment]
selectdropdown

**Comment by pkozlowski-opensource** [comment]
Woooho, thnx @mohammedzamakhan !

@wesleycho @Foxandxss @maxokorokov @jnizet what are you thoughts guys? I don't think we've planned on having this directive initially but why not. @mohammedzamakhan do you think you would have time to maintain this component if it is merged? I guess my main fear is that people will love the directive so will open issues / feature requests :-) Not the worst problem to have but challanging if you are already stretched with maintenance.

**Comment by mohammedzamakhan** [comment]
I would love to maintain this component @pkozlowski-opensource 👍

**Comment by Foxandxss** [comment]
I will take a peek

File: src/select/select.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+   */
+  @Input() ngbLabel = 'label';
+
+  /**
+   * Value property name
+   */
+  @Input() ngbValue = 'value';
+
+  onChange = (_: any) => {};
+  onTouched = () => {};
```
Comment (wesleycho) [inline src/select/select.ts]: Are these functions to be implemented?

File: src/select/select.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  registerOnChange(fn: (value: any) => any): void { this.onChange = fn; }
+
+  registerOnTouched(fn: () => any): void { this.onTouched = fn; }
```
Comment (wesleycho) [inline src/select/select.ts]: I see these registration functions, but there does not seem to be a way to obviously use these - do these require one to use the exportAs feature and do `ngbSelect.registerOnChange(...)` in the template?

**Comment by wesleycho** [comment]
This mostly LGTM, just some questions about API usage for registering onChange and onTouched & would like to see examples using it.

**Comment by e-globa** [comment]
Would this component be editable? I mean, as an user can I type something into input for filtering values in a list?

**Comment by mohammedzamakhan** [comment]
@e-globa yes we can add a search input field in the list options for filtering

**Comment by e-globa** [comment]
Hi, @mohammedzamakhan.
When is it expected to be released?

**Comment by ASHFAQPATWARI** [comment]
This would be a nice addition to library. @mohammedzamakhan Any update ?

File: src/select/select.ts (hunk 2 above)
Comment (CSchulz) [inline src/select/select.ts]: Needed for the ValueAccessor.

File: src/select/select.ts (hunk 1 above)
Comment (CSchulz) [inline src/select/select.ts]: They got automatically registered with the registerX methods by the form module.

**Comment by kotmatpockuh** [comment]
any news?

**Comment by marcelnem** [comment]
It seems that @mohammedzamakhan probably does not have time for this anymore. Is there any future for the select? Is it on the roadmap or can anybody take over?

**Comment by mohammedzamakhan** [comment]
@marcelnem I got busy with a project on React, so wasn't able to focus on this, but soon I will get back to Angular project (in couple of weeks), and I hope I can work on this component support

**Comment by sergeu90** [comment]
Hellow . Is there any news?

**Comment by FloProg** [comment]
I would be very happy if this feature could be added. Is this going to support the "multiple" attribute and offers checkboxes to select multiple options? The standard select multiple of bootstrap is very big, not very nice nor very user-friendly in my opinion.

**Comment by pkozlowski-opensource** [comment]
We would very much like to have this type of widget but in reality we don't have free cycles  at the moment to support such a non-trivial widget. We will keep an eye on it via #2371 but can't afford to incorporate  and maintain this new chunk of code.

**Pull request closed without merging**